src/index.js

```diff
@@ -1,14 +1,25 @@
-import React from 'react';
-import ReactDOM from 'react-dom/client';
-import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import React from "react";
+import ReactDOM from "react-dom/client";
+import "./index.css";
+import App from "./App";
+import reportWebVitals from "./reportWebVitals";
+import { configureStore } from "@reduxjs/toolkit";
+import { Provider } from "react-redux";
+import todoReducer from "./features/todo";
 
-const root = ReactDOM.createRoot(document.getElementById('root'));
+const store = configureStore({
+    reducer: {
+        todo: todoReducer,
+    },
+});
+
+const root = ReactDOM.createRoot(document.getElementById("root"));
 root.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
+    <Provider store={store}>
+        <React.StrictMode>
+            <App />
+        </React.StrictMode>
+    </Provider>
```

Comment:
StrictMode was made to help us to catch hidden errors, if you do not wrap the whole project this will be bad for your app. 
here the strict mode will not be used on the store file
```suggestion
    <React.StrictMode>
             <Provider store={store}>
             <App />
             </Provider>
        </React.StrictMode>
```